docarray/doc_index/backends/elastic_doc_index.py

```diff
@@ -0,0 +1,454 @@
+            if not col.config:
+                continue  # do not create column index if no config is given
+            body['mappings']['properties'][col_name] = self._create_index(col)
+
+        if self._client.indices.exists(index=self._index_name):  # type: ignore
+            self._client.indices.put_mapping(
+                index=self._index_name, body=body['mappings']
+            )
+        else:
+            self._client.indices.create(index=self._index_name, body=body)
+
+        if len(self._db_config.index_settings):
+            self._client.indices.put_settings(
+                index=self._index_name, body=self._db_config.index_settings
+            )
+
+        self._refresh(self._index_name)
+
+    ###############################################
+    # Inner classes for query builder and configs #
+    ###############################################
+
+    class QueryBuilder(BaseDocumentIndex.QueryBuilder):
+        def __init__(self, outer_instance, **kwargs):
+            super().__init__()
+            self._search = Search(
+                using=outer_instance._client, index=outer_instance._index_name, **kwargs
+            )
+
+        def build(self, *args, **kwargs) -> Any:
+            return self
+
+        def query(self, *args, **kwargs):
+            self._search = self._search.query(*args, **kwargs)
+            return self
+
+        def filter(self, *args, **kwargs):
+            self._search = self._search.filter(*args, **kwargs)
+            return self
+
+        def exclude(self, *args, **kwargs):
+            self._search = self._search.exclude(*args, **kwargs)
+            return self
+
+        def script_fields(self, **kwargs):
+            self._search = self._search.script_fields(**kwargs)
+            return self
+
+        def sort(self, *args, **kwargs):
+            self._search = self._search.sort(*args, **kwargs)
+            return self
+
+        # TODO: add more methods
+
+        find = _raise_has_substitue('find')
+        text_search = _raise_has_substitue('text_search')
+        find_batched = _raise_not_composable('find_batched')
+        filter_batched = _raise_not_composable('find_batched')
+        text_search_batched = _raise_not_composable('text_search')
+
+    def build_query(self, **kwargs) -> QueryBuilder:
+        """
+        Build a query for this DocumentIndex.
+        """
+        return self.QueryBuilder(self, **kwargs)  # type: ignore
+
+    @dataclass
+    class DBConfig(BaseDocumentIndex.DBConfig):
+        hosts: Union[str, List[str], None] = 'http://localhost:9200'
+        index_name: Optional[str] = None
+        es_config: Dict[str, Any] = field(default_factory=dict)
+        index_settings: Dict[str, Any] = field(default_factory=dict)
+
+    @dataclass
+    class RuntimeConfig(BaseDocumentIndex.RuntimeConfig):
+        default_column_config: Dict[Type, Dict[str, Any]] = field(
+            default_factory=lambda: {
+                np.ndarray: {
```

Comment:
Would be nice to also handle torch and tensorflow, maybe also list of numbers etc